site/src/common/blocks/DamVideoBlock.tsx

```diff
@@ -1,18 +1,59 @@
 import { PreviewSkeleton, PropsWithData, withPreview } from "@comet/cms-site";
-import { DamVideoBlockData } from "@src/blocks.generated";
-import * as React from "react";
+import { MediaDamVideoBlockData } from "@src/blocks.generated";
+import { VideoPreviewImage } from "@src/common/helpers/VideoPreviewImage";
+import { useState } from "react";
+import styled, { css } from "styled-components";
+
+interface DamVideoBlockProps extends PropsWithData<MediaDamVideoBlockData> {
+    sizes?: string;
+}
 
 export const DamVideoBlock = withPreview(
-    ({ data: { damFile, autoplay, showControls } }: PropsWithData<DamVideoBlockData>) => {
+    ({ data: { video, aspectRatio, previewImage }, sizes = "100vw" }: DamVideoBlockProps) => {
+        const { damFile, autoplay, loop, showControls } = video;
+
         if (damFile === undefined) {
             return <PreviewSkeleton type="media" hasContent={false} />;
         }
 
+        const hasPreviewImage = previewImage && previewImage.block?.props.damFile;
+        const [showPreviewImage, setShowPreviewImage] = useState(true);
+
         return (
-            <video autoPlay={autoplay} controls={showControls} playsInline muted={autoplay}>
-                <source src={damFile.fileUrl} type={damFile.mimetype} />
-            </video>
+            <>
+                {hasPreviewImage && showPreviewImage && (
+                    <VideoPreviewImage
+                        onClick={() => setShowPreviewImage(false)}
+                        image={previewImage}
+                        aspectRatio={aspectRatio !== "auto" ? aspectRatio : undefined}
+                        sizes={sizes}
+                    />
+                )}
+                {(!showPreviewImage || !hasPreviewImage) && (
```

Comment:
These conditions can be combined

```suggestion
                {hasPreviewImage && showPreviewImage ?(
                    <VideoPreviewImage
                        onClick={() => setShowPreviewImage(false)}
                        image={previewImage}
                        aspectRatio={aspectRatio !== "auto" ? aspectRatio : undefined}
                        sizes={sizes}
                    />
                ) : (
```